### ISSUE TITLE
Lemmatization of contraction "we're/she's" does not yield pronoun + "be"

### ISSUE THREAD

**Issue by henningko** [explosion/spaCy#321]
I'm currently parsing a lot of speech transcripts, encountering a lot of pronoun contractions with subject + be. While `I'm` will be lemmatized into `I be`, this does not happen for contractions like `you're` or `he's`, the lemmas for the latter will actually be `he` and `'`.

Is there a reason for treating contractions this way?

**Comment by honnibal** [comment]
Hmm. Those contractions should all be handled correctly --- the intention
is definitely to have these handled as "be". Could this be a smart quotes
issue?

The list of special cases can be found in lang_data/en/specials.json. It's
keyed by a string, which maps to one or more token descriptions.

**Comment by henningko** [comment]
Thanks for the reply. Double checked in shell, not a problem with smart quotes.
```python
import spacy.en

nlp = spacy.en.English()
doc = nlp("I'm not sure. They're sure.")  # random sentence
lemmata = [token.lemma_ for token in doc]
```
yields `['i', 'be', 'not', 'sure', '.', 'they', "'re", 'sure', '.']`

I'm on OS X 10.11.4, running conda with Python 3.5.1 and spaCy 0.100.6.

**Comment by honnibal** [comment]
Okay, thanks. I should've looked more closely. There were indeed several remaining inconsistencies in the script that generates the specials.json file. I'm surprised this didn't get spotted sooner!

I still need to add some tests for this, but the issue should be fixed in the next data release.

**Comment by henningko** [comment]
Absolute python newbie question, but how can I install the latest GitHub version if I've installed spaCy through conda?
Tried `pip install --upgrade git+https://github.com/spacy-io/spaCy.git`, but that fails with some Cython errors.

**Comment by henningpeters** [comment]
Please check out https://spacy.io/docs#install-source

Pro tip: you can install with pip's git+https handler if you ran `pip install cython`.

**Comment by honnibal** [comment]
There's currently an issue with Cython 0.24, so make sure you're using 0.23.

Also, note that the lemmatization is controlled in a data file. So, even once you get spaCy compiled, it won't behave any differently if you use the default data. The generate_specials.py file creates a file specials.json, which by default will be at data/en/tokenizer/specials.json within the spacy package.

**Comment by henningko** [comment]
Hmm, couldn't compile with cython 0.23.5 (and cython 0.23.1)
```
pip install -e .
Obtaining file:///Users/henningk/spaCy
    Complete output from command python setup.py egg_info:

    Error compiling Cython file:
    ------------------------------------------------------------
    ...
    from cymem.cymem cimport Pool
    from thinc.typedefs cimport weight_t
    ^
    ------------------------------------------------------------

    syntax/transition_system.pxd:2:0: 'thinc/typedefs.pxd' not found
```

Thinc is installed. Weird, b/c installing via Conda works fine.

Anyways, this is going off-topic. Appreciate both of you helping!

@honnibal: gotcha, should've checked the file out better. Will just generate the specials.json then.

**Comment by henningpeters** [comment]
As this is becoming increasingly off-topic please open a new issue with instructions how to reproduce if that error is still present.

**Comment by henningko** [comment]
A few more inconsistencies:
```python
>>> doc = nlp("I'm going home. I am going home. He's going home. He is going home.")
>>> print(' '.join([word.lemma_ for word in doc]))
-PRON- be go home . i be go home . -PRON- ' go home . he be go home .
```
Essentially, any pronoun that's in a contraction will get lemmatized to `-PRON-`, rather than staying untouched (which I feel would be the intended behavior).
Also, `he's` still lemmatizes to `'` (it had not been touched in 85485f5). As `'s` can refer to a contraction of be or to a possessive, maybe it's worthwhile to check the POS-tagging result, and lemmatize if the tag is `VBZ` (as far as I can tell, the tagger works correctly)? This would probably have to be addressed outside of the specials.json.

**Comment by honnibal** [comment]
These should be fixed in 1.0.

**Comment by transfluxus** [comment]
I've noticed that contractions are not contractions are resolved but not detected as stop words.
For example:
```
doc = nlp("You're cool. You are cool")
for token in doc:
    if not (token.is_stop or token.is_punct):
        print(token, token.lemma_, token.pos_, token.tag_)
```
results in:
```
're be VERB VBP
cool cool ADJ JJ
cool cool ADJ JJ
```
Is that intended or efficiently resolved? other then
```
doc = nlp("You're cool. You are cool")
for token in doc:
    if not (token.is_stop or token.is_punct or token.text[0] == "'"):
        print(token, token.lemma_, token.pos_, token.tag_)
```
